sklearn/cross_decomposition/_pls.py

```diff
@@ -221,7 +221,7 @@ def fit(self, X, Y):
             Training vectors, where `n_samples` is the number of samples and
             `n_features` is the number of predictors.
 
-        Y : array-like of shape (n_samples,) or (n_samples, n_targets)
+        y : array-like of shape (n_samples,) or (n_samples, n_targets)
             Target vectors, where `n_samples` is the number of samples and
             `n_targets` is the number of response variables.
 
```

Comment:
You need to add a docsring for the `Y` parameter and add the `.. deprecated::` info. You can check the following documentation for more details: https://scikit-learn.org/dev/developers/contributing.html#deprecation